pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -262,20 +263,28 @@ func (r *ReconcilePerconaServerMongoDB) Reconcile(request reconcile.Request) (re
 		}
 	}
 
-	for i, replset := range cr.Spec.Replsets {
-		// multiple replica sets is not supported until sharding is
-		// added to the operator
-		if i > 0 {
-			reqLogger.Error(nil, "multiple replica sets is not yet supported, skipping replset %s", replset.Name)
+	repls := cr.Spec.Replsets
+	if cr.Spec.Sharding.Enabled && cr.Spec.Sharding.ConfigsvrReplSet != nil {
+		repls = append(repls, cr.Spec.Sharding.ConfigsvrReplSet)
+	}
+
+	for i, replset := range repls {
+		if !cr.Spec.Sharding.Enabled && i > 0 {
+			reqLogger.Error(nil, "multiple replica sets is not supported for non sharded configuration, skipping replset %s", replset.Name)
 			continue
 		}
 
+		if replset.ClusterRole != api.ClusterRoleConfigSvr && replset.Name == "cfg" {
+			return reconcile.Result{}, errors.New("cfg is reserved name for config server replset")
+		}
```

Comment:
Should we care if `.Name` option used to name `configsvrReplSet` other than "cfg"?

Reply:
Not sure. I think we can allow to give any name.

Reply:
I like @dAdAbird idea 
```suggestion
		if replset.ClusterRole != api.ClusterRoleConfigSvr && cr.Spec.Sharding != nil && cr.Spec.Sharding.Enabled && cr.Spec.Sharding.ConfigsvrReplSet != nil && replset.Name == cr.Spec.Sharding.ConfigsvrReplSet.Name {
			return reconcile.Result{}, errors.New("cfg is reserved name for config server replset")
		}
```

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/439/commits/14d3a82f1ba8b982fe48937f9fbe6322677acc42